### PR TITLE
LRU

### PR DESCRIPTION
**Pull request by wilsonyqm**

### CODE REVIEW

**Comment by wilsonyqm** [comment]
[LRU cache](https://trello.com/c/NnfphCqd/1-lru-cache)

File: LRU0.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+public class Solution {
+
+    // @param capacity, an integer
+    class Node{
```
Comment (Ethan199111) [inline LRU0.java:4]: it is rare to declare your class as default package-private.

File: LRU0.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        int key, value;
```
Comment (Ethan199111) [inline LRU0.java:5]: final int key, cacheKey cannot be modified.

File: LRU0.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        Node prev, next;
+        Node(int k, int v){
```
Comment (Ethan199111) [inline LRU0.java:7]: same for constructor

File: LRU0.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            key = k;
+            value = v;
+            next = null;
+            prev = null;
+        }
+    }
+    HashMap<Integer,Node> map;
```
Comment (Ethan199111) [inline LRU0.java:14]: collections can also be final, it means reference cannot be modified

File: LRU0.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    private int capacity;
+    private Node tail, head;
+    public Solution(int capacity) {
+        this.capacity = capacity;
+        map = new HashMap<>();
+        head = new Node(-1, -1);
```
Comment (Ethan199111) [inline LRU0.java:20]: use static instead of creating new instance

File: LRU0.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            node.prev.next = node.next;
+            node.next.prev = node.prev;
+        }
+        node.next = head.next;
+        head.next.prev = node;
+        node.prev = head;
+        head.next = node;
+    }
+    // @return an integer
+    public int get(int key) {
```
Comment (Ethan199111) [inline LRU0.java:38]: Use private since this method can only be accessed by LRUCache.class. Use private methods unless you need to override a public API

**Pull request merged**